ts/table-grid/table-grid.controller.ts

```diff
@@ -150,6 +151,10 @@ module Lui.Directives {
 				return tmp.textContent || tmp.innerText || "";
 			};
 
+			$scope.updateCheckboxes = () => {
+				_.each($scope.datas, (data: any) => { data.isChecked = $scope.allChecked.value; });
+			};
```

Comment:
maybe something like 
```
$scope.onMasterCheckBoxesChange: () => {
    if (_.some(rows, (row) => { !row.isChecked; })
    {
        _.each(rows, (row) => { row.isChecked = true; })
    } else {
        _.each(rows, (row) => { row.isChecked = false; })
    }
}
```

You don't need $scope.allChecked anymore plus you can uncked everything with a double click on the master checkbox.